crates/uv-resolver/src/lock.rs

```diff
@@ -236,6 +247,52 @@ impl Lock {
             .expect("valid index for distribution");
         dist
     }
+
+    /// Convert the [`Lock`] to a [`InMemoryIndex`] that can be used for resolution.
+    pub fn to_index(self, workspace: &Workspace) -> Result<InMemoryIndex, LockError> {
+        let distributions = FxOnceMap::default();
+        let mut packages: HashMap<_, BTreeMap<Version, PrioritizedDist>> = HashMap::new();
+
+        for distribution in self.distributions {
+            match distribution.id.source {
+                Source::Registry(..) | Source::Git(..) => {}
+                // Skip local and direct URL dependencies, as their metadata may have been mutated
+                // without a version change.
+                Source::Path(..)
+                | Source::Directory(..)
+                | Source::Editable(..)
+                | Source::Direct(..) => continue,
+            }
+
+            // Add registry distributions to the package index.
+            if let Some(prioritized_dist) = distribution.to_prioritized_dist(workspace.root())? {
+                packages
+                    .entry(distribution.name().clone())
+                    .or_default()
+                    .insert(distribution.id.version.clone(), prioritized_dist);
+            }
+
+            // Extract the distribution metadata.
+            let version_id = distribution.version_id(workspace.root())?;
+            let hashes = distribution.hashes();
+            let metadata = distribution.into_metadata(workspace.root())?;
+
+            // Add metadata to the distributions index.
+            let response = MetadataResponse::Found(ArchiveMetadata::with_hashes(metadata, hashes));
+            distributions.done(version_id, Arc::new(response));
+        }
+
+        let packages = packages
+            .into_iter()
+            .map(|(name, versions)| {
+                // TODO(ibraheem): should all distributions be in the same `VersionMap`?
```

Comment:
I think this is correct. The `Vec<VersionMap>` seems to be used when fetching from multiple indexes, but they are all merged eventually anyways. I think it could that distinction could be removed.